### ISSUE TITLE
EurekaHealthIndicator  - unreachable condition

### ISSUE THREAD

**Issue by deripas** [spring-cloud/spring-cloud-netflix#3058]
In `EurekaHealthIndicator` unreachable condition
``` java
	private Status getStatus(Builder builder) {
		Status status = new Status(
				this.eurekaClient.getInstanceRemoteStatus().toString(),
				"Remote status from Eureka server");

		if (eurekaClient instanceof DiscoveryClient && clientConfig.shouldFetchRegistry()) {
			DiscoveryClient discoveryClient = (DiscoveryClient) eurekaClient;
			long lastFetch = discoveryClient.getLastSuccessfulRegistryFetchTimePeriod();

			if (lastFetch < 0) {
				status = new Status("UP",
						"Eureka discovery client has not yet successfully connected to a Eureka server");
			}
			else if (lastFetch > clientConfig.getRegistryFetchIntervalSeconds() * 2000) {
				status = new Status("UP",
						"Eureka discovery client is reporting failures to connect to a Eureka server");
				builder.withDetail("renewalPeriod",
						instanceConfig.getLeaseRenewalIntervalInSeconds());
				builder.withDetail("failCount",
						lastFetch / clientConfig.getRegistryFetchIntervalSeconds());
			}
		}

		return status;
	}

```
 `eurekaClient instanceof DiscoveryClient`  always **false**!
`eurekaClient` is  `org.springframework.cloud.netflix.eureka.CloudEurekaClient@3d369e4b`  instanceof `EurekaClient`, but **NOT** `DiscoveryClient`.

And why do you need a class `EurekaHealthIndicator`? In my opinion it is useless.
PS: version 2.0.0-RELEASE

**Comment by ryanjbaxter** [comment]
`CloudEurekaClient` extends `DiscoveryClient` so that should be `true`.

**Comment by deripas** [comment]
yes ofcource, but in runtime `eurekaClient ` not realy object, it's jdk proxy.
`DiscoveryClient` class, not interface. Proxy `eurekaClient ` can't cast to DiscoveryClient.

![image](https://user-images.githubusercontent.com/14219705/42477122-ade65bce-83d8-11e8-99c5-8f580aa1da8c.png)

![image](https://user-images.githubusercontent.com/14219705/42477064-87572538-83d8-11e8-9807-d1f020fdd369.png)

**Comment by ryanjbaxter** [comment]
In my test it was an instance.  Can you provide a use case/sample where you can replicate this?

**Comment by deripas** [comment]
Create [simple project](https://github.com/deripas/simple-eureka-client) with Eureka client and Actuator . For example, in `SimpleEurekaClientApplication` class inject `eurekaClient`:
```java
    @Autowired
    private EurekaClient eurekaClient;

    @PostConstruct
    public void init() {
      log.warn("eurekaClient class: {}", eurekaClient.getClass().getCanonicalName());
      log.warn("eurekaClient instanceof DiscoveryClient: {}", eurekaClient instanceof DiscoveryClient);
    }
```
Result output:
```log
2018-07-10 12:52:23.744  WARN 21767 --- [           main] c.e.s.e.c.SimpleEurekaClientApplication  : eurekaClient class: com.sun.proxy.$Proxy91
2018-07-10 12:52:23.744  WARN 21767 --- [           main] c.e.s.e.c.SimpleEurekaClientApplication  : eurekaClient instanceof DiscoveryClient: false
```

**Comment by ryanjbaxter** [comment]
Fixed via 42449ee
